lib/common/alarm/AlarmFactory.ts

```diff
@@ -821,9 +821,37 @@ export class AlarmFactory {
     };
   }
 
+  /**
+   * Creates a composite alarm from a collection of alarms and composite alarms.
+   *
+   * @param alarms Array of individual alarms to be composed
+   * @param props Customization options for the composite alarm
+   * @param compositeAlarms Optional array of composite alarms to be composed together with regular alarms
+   * @returns Newly created composite alarm
+   */
   addCompositeAlarm(
     alarms: AlarmWithAnnotation[],
     props: AddCompositeAlarmProps,
+    compositeAlarms: CompositeAlarm[] = [],
+  ): CompositeAlarm {
+    const alarmBases = [
+      ...alarms.map((alarm) => alarm.alarm),
+      ...compositeAlarms,
+    ] as AlarmBase[];
```

Comment:
Is this cast necessary? I thought `CompositeAlarm` was already an `AlarmBase`.

Reply:
yeah, it's not needed, it was mostly just a visual cue. I can remove it.

Reply:
We can specify the type for the variable instead in that case (e.g. `const alarmBases: AlarmBase[] = [...];`.

Reply:
Yeah, I removed it. Thinking about it, it's not really needed.